spec/datadog/tracing/contrib/utils/quantization/http_spec.rb

```diff
@@ -271,6 +278,101 @@
           end
         end
       end
+
+      context 'and an obfuscate: :internal option' do
+        context 'with a non-matching substring' do
+          let(:query) { 'foo=foo' }
+          let(:options) { { obfuscate: :internal } }
+
+          it { is_expected.to eq('foo=foo') }
+        end
+
+        context 'with a matching substring at the beginning' do
+          let(:query) { 'pass=03cb9f67-dbbc-4cb8-b966-329951e10934&key2=val2&key3=val3' }
+          let(:options) { { obfuscate: :internal } }
+
+          it { is_expected.to eq('<redacted>&key2=val2&key3=val3') }
+        end
+
+        context 'with a matching substring in the middle' do
+          let(:query) { 'key1=val1&public_key=MDNjYjlmNjctZGJiYy00Y2I4LWI5NjYtMzI5OTUxZTEwOTM0&key3=val3' }
+          let(:options) { { obfuscate: :internal } }
+
+          it { is_expected.to eq('key1=val1&<redacted>&key3=val3') }
+        end
+
+        context 'with a matching substring at the end' do
+          let(:query) { 'key1=val1&key2=val2&token=03cb9f67dbbc4cb8b966329951e10934' }
+          let(:options) { { obfuscate: :internal } }
+
+          it { is_expected.to eq('key1=val1&key2=val2&<redacted>') }
+        end
+
+        context 'with a matching, URL-encoded JSON substring' do
+          let(:query) { 'json=%7B%20%22sign%22%3A%20%22%7B0x03cb9f67%2C0xdbbc%2C0x4cb8%2C%7B0xb9%2C0x66%2C0x32%2C0x99%2C0x51%2C0xe1%2C0x09%2C0x34%7D%7D%22%7D' }
+          let(:options) { { obfuscate: :internal } }
+
+          it { is_expected.to eq('json=%7B%20%22<redacted>%7D') }
```

Comment:
Yup, this produces invalid JSON: `"{ \"<redacted>}"`

Reply:
Not sure we can address this right now. I'm ok with this specific edge case at this point.

Reply:
Addressed the stray quote by updating the regex.